.packit.yaml

```diff
@@ -14,198 +14,142 @@ srpm_build_deps:
-            - full test
-          absent:
-            - discuss
     tf_extra_params:
-        test:
-            tmt:
-                name: '/plans/features/extended-unit-tests$'
-
-  # Test pull requests (provision)
-  - job: tests
-    identifier: provision
-    trigger: pull_request
+      test:
+        tmt:
+          name: '/plans/features/extended-unit-tests$'
+  - # Test pull requests (provision)
+    <<: *tests
     use_internal_tf: True
-    targets:
+    targets: &internal-targets
       - fedora-latest-stable
+    require: *require-full-tests
+    identifier: provision
     tf_extra_params:
-        test:
-            tmt:
-                name: /plans/provision/virtual
-        environments:
-          - tmt:
-                context:
-                    how: provision
-    require:
-        label:
-          present:
-            - full test
-          absent:
-            - discuss
-
-  # Test internal plugins
-  - job: tests
-    trigger: pull_request
-    identifier: "internal-plugins"
-    targets:
-      - fedora-latest-stable
+      test:
+        tmt:
+          name: /plans/provision/virtual
+      environments:
+        - tmt:
+            context:
+              how: provision
+  - # Test internal plugins
+    <<: *tests
     use_internal_tf: True
+    targets: *internal-targets
+    require: *require-full-tests
+    identifier: internal-plugins
     fmf_url: "https://gitlab.cee.redhat.com/baseos-qe/tmt.git"
     # Tag cloud resources for tmt
     tf_extra_params:
-        environments:
-          - settings:
-                provisioning:
-                    tags:
-                        BusinessUnit: tmt
-    require:
-        label:
-          present:
-            - full test
-          absent:
-            - discuss
-
-  # Test internal wow
-  - job: tests
-    trigger: pull_request
-    identifier: "internal-wow"
-    targets:
-      - fedora-latest-stable
+      environments:
+        - &tmt-cloud-resources
+          settings:
+            provisioning:
+              tags:
+                BusinessUnit: tmt
+  - # Test internal wow
+    <<: *tests
     use_internal_tf: True
+    targets: *internal-targets
+    require: *require-full-tests
+    identifier: internal-wow
     fmf_url: "https://gitlab.cee.redhat.com/baseos-qe/integration_scripts.git"
     tmt_plan: "/tmt/integration/plan"
     tf_extra_params:
-        environments:
-          - settings:
-                provisioning:
-                    tags:
-                        BusinessUnit: tmt
-    require:
-        label:
-          present:
-            - full test
-          absent:
-            - discuss
+      environments:
+        - *tmt-cloud-resources
 
   # Build commits to main
-  - job: copr_build
+  - <<: *copr_build
     trigger: commit
     branch: main
-    targets:
-      - fedora-all
-      - epel-9
-    enable_net: False
     list_on_homepage: True
     preserve_project: True
     owner: "@teemtee"
     project: tmt
```

Comment:
What I want to open a discussion on is where/how [`@teemtee/tmt`](https://copr.fedorainfracloud.org/coprs/g/teemtee/tmt/builds/) should work, particularly w.r.t. `release_suffix` and `get-current-version`. Particularly trigger `commit` (the latest commit) will always win since it is bumped by hatch version bumping the version above the `release` triggered. This is different than using just the `git describe` that is `packit` default

Initially I thought that `copr` has some special sauce to always serve only the latest copr build of a project, but lately I've found the hard way that that is not true and it takes the NVR as per normal package installation. Not sure about how long a build is available though.

---

So this raises the question what is the intended purpose of `@teemtee/tmt`, and what are some alternative. Currently there are 3 `copr_build` jobs, which effectively contain:
```yaml
  - <<: *copr_build
    trigger: pull_request
    actions:
      <<: *base-actions
      get-current-version:
        - bash -c "hatch version | sed -E 's/\\.[0-9]+\\.dev.*/.dev888/'"
  - <<: *copr_build
    trigger: commit
    branch: main
    owner: "@teemtee"
    project: tmt
    actions:
      <<: *base-actions
      get-current-version:
        - bash -c "hatch version | sed -E 's/\\.[0-9]+\\.dev/.dev/'"
  - <<: *copr_build
    trigger: release
    owner: "@teemtee"
    project: tmt
    actions:
      <<: *base-actions
      get-current-version:
        - hatch version
```
- *pull_request*: This gets built in a unique copr project, e.g. `packit/teemtee-tmt-3020`, so there is never an issue with clashing builds if one wants to install it locally (temporarily).
  But then the `get-current-version` here is rather meaningless. Maybe the default can be used instead or tweak it a bit with `release_suffix`?
- *commit* and *release*: These are built on the same `@teemtee/tmt` project, and because of hatch bumping the version every time, only `commit` would be installable or pulled by other copr additional_dependencies.
  How about instead to create `@teemtee/release` where all release builds are stored and have a `@teemtee/tmt` and `@teemtee/fmf` act like nightly builds? In both cases though, `get-current-version` could be replaced with the default version construct? It's more of a preference on how it could display on the copr project since both cases should have equivalent preference over older builds in the commit history.

Reply:
@martinhoyer @lukaszachy @psss any comments on this?

Reply:
Not sure how, but 'release' as https://copr.fedorainfracloud.org/coprs/g/teemtee/tmt/build/7628108/ are present in the teemtee/tmt repo.

Reply:
> Not sure how, but 'release' as https://copr.fedorainfracloud.org/coprs/g/teemtee/tmt/build/7628108/ are present in the teemtee/tmt repo.

Yes, that's the current behavior, both `release` and `commit` are sharing the `@teemtee/tmt`, and what I want to question here, is that really a good approach? The user is unable to choose if they want the `nightly` or the `release` (maybe for before it lands in Fedora, but also we could expand it to other distros?). With how the versioning works, `@teemtee/tmt` is effectively `nightly`.

Another benefit of using `release` and `nightly` is that it can be shared with `fmf` and we can more easily test some coupled changes.

Reply:
> Yes, that's the current behavior, both `release` and `commit` are sharing the `@teemtee/tmt`, and what I want to question here, is that really a good approach? The user is unable to choose if they want the `nightly` or the `release`

I'd say the choice is there, in Fedora versus Copr:

 * if you want the latest greatest bleeding edge you enable copr
 * if you're ok with the released versions you stick with the default Fedora repos

> The shared space with fmf seems tempting though.

That resonates with me as well. Having a single repo where the latest `fmf` and `tmt` bits would be available together sounds useful. Users would have to enabled just a single repo.

Reply:
+1 for `@teemtee/nightly` with tmt and fmf builds!  
Let's do that in separate Issue/PR though :) I'd be happy to work on this with you @LecrisUT, if you let me ;)  

Also, if we could find some more elegant way to deal with the dev888 versions, I'd be able to sleep well again lol.

Reply:
> +1 for `@teemtee/nightly` with tmt and fmf builds!

Ack. What about calling it `@teemtee/latest`? It would be nicely aligned with the latest documentation:

* https://tmt.readthedocs.io/en/latest/

> Let's do that in separate Issue/PR though :)

Yeah, let's cover this change separately and not block this pull request on this.

Reply:
:+1: If we manage to go the separate copr projects, the whole `hatch` version would not be necessary anymore :wink:. We will lose the lucky numbers 888 :P